crates/ty_python_semantic/src/module_resolver/resolver.rs

```diff
@@ -38,7 +56,41 @@ pub(crate) fn resolve_module_query<'db>(
     let name = module_name.name(db);
     let _span = tracing::trace_span!("resolve_module", %name).entered();
 
-    let Some(resolved) = resolve_name(db, name) else {
+    let Some(resolved) = resolve_name(db, name, ModuleResolveMode::StubsAllowed) else {
+        tracing::debug!("Module `{name}` not found in search paths");
+        return None;
+    };
+
+    let module = match resolved {
+        ResolvedName::FileModule(module) => {
+            tracing::trace!(
+                "Resolved module `{name}` to `{path}`",
+                path = module.file.path(db)
+            );
+            Module::file_module(name.clone(), module.kind, module.search_path, module.file)
+        }
+        ResolvedName::NamespacePackage => {
+            tracing::trace!("Module `{name}` is a namespace package");
+            Module::namespace_package(name.clone())
+        }
+    };
+
+    Some(module)
+}
+
+/// Salsa query that resolves an interned [`ModuleNameIngredient`] to a module.
+///
+/// This query should not be called directly. Instead, use [`resolve_module`]. It only exists
+/// because Salsa requires the module name to be an ingredient.
+#[salsa::tracked(heap_size=get_size2::GetSize::get_heap_size)]
+pub(crate) fn resolve_real_module_query<'db>(
+    db: &'db dyn Db,
+    module_name: ModuleNameIngredient<'db>,
+) -> Option<Module> {
```

Comment:
I duplicated this in case I'd need to tweak the behaviour, but I never really did. I can/should just make resolve_module_query take two arguments.